enterprise/cli/licenses_test.go

```diff
@@ -0,0 +1,182 @@
+package cli_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/cli/clitest"
+	"github.com/coder/coder/coderd/coderdtest"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/enterprise/cli"
+	"github.com/coder/coder/enterprise/coderd"
+	"github.com/coder/coder/pty/ptytest"
+	"github.com/coder/coder/testutil"
+)
+
+const fakeLicenseJWT = "test.jwt.sig"
+
+func TestLicensesAddSuccess(t *testing.T) {
+	t.Parallel()
+	// We can't check a real license into the git repo, and can't patch out the keys from here,
+	// so instead we have to fake the HTTP interaction.
+	t.Run("LFlag", func(t *testing.T) {
+		t.Parallel()
+		cmd := setupFakeLicenseServerTest(t, "licenses", "add", "-l", fakeLicenseJWT)
+		pty := attachPty(t, cmd)
+		errC := make(chan error)
+		go func() {
+			errC <- cmd.Execute()
```

Comment:
Suggestion: Consider using test timeouts (`ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)`) coupled with `cmd.ExecuteContext(ctxThatTimesOut)`.

This is to avoid situations where a failing tests (due to whatever circumstance) causes the suite to time out, producing convoluted failure output.